### PR TITLE
Automation: RadioGroup: Expose isDisabled slot prop

### LINKED ISSUE #8977: [@rancher/components] RadioGroup.vue: Expose disabled slot prop for indexed slots
Exposing a slot prop to communicate when the radio group is disabled will be helpful for tracking the overall disabled state when composing components.

An example of this usage can be seen in rancher-sandbox/rancher-desktop#4758

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #8977

### Technical notes summary
Added test using an example wrapper to integrate slot case.

### CODE REVIEW

File: pkg/rancher-components/src/components/Form/Radio/RadioGroup.test.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { mount } from '@vue/test-utils';
+import { RadioGroup } from './index';
+
+describe('component: RadioGroup', () => {
+  describe('when disabled', () => {
+    /**
+     * NOTE: Current setup does not match guidelines for the test of scoped slots
```
Comment (rak-phillip) [inline pkg/rancher-components/src/components/Form/Radio/RadioGroup.test.ts:7]: This test looks very similar to the recommendation for v1 of Vue Test Utils, is there something that I'm missing in this note?

https://v1.test-utils.vuejs.org/api/options.html#scopedslots

**Review by rak-phillip — COMMENTED** [review]
I think this change looks great, I just have a question about the note before the test.

File: pkg/rancher-components/src/components/Form/Radio/RadioGroup.test.ts (hunk 1 above)
Comment (cnotv) [reply pkg/rancher-components/src/components/Form/Radio/RadioGroup.test.ts:7]: That's a very good point, I did not notice that they changed also this.

**Review by rak-phillip — APPROVED** [review]
LGTM

**Pull request merged**